### ISSUE TITLE
cmd/go: script test hanging with blocked output pipes

### ISSUE THREAD

**Issue by bcmills** [golang/go#53400]
`greplogs -l -e 'goroutine \d+ \[chan receive, \d+ minutes\]:\n(?:.+\n\t.+\n)*cmd/go_test\.waitOrStop' --since=2022-01-01`
[2022-06-06T15:42:19-47e34ca/windows-amd64-longtest](https://build.golang.org/log/351333343bb6c9f6ef30e2da29488efe212896a4)
[2022-05-26T22:25:23-55c42c2/windows-amd64-longtest](https://build.golang.org/log/658ec778c98a6bca716e76d3ae00fc9d0eac92ad)
[2022-05-20T23:37:43-0a1a092/linux-amd64-longtest](https://build.golang.org/log/35b29b457464ef4271b9576dbcf3b94fb101dd33)
[2022-05-20T23:37:43-0a1a092/windows-amd64-longtest](https://build.golang.org/log/47aab0e3e3e6a010c78eafa3f50e72605054538d)
[2022-05-20T23:05:38-c3470ca/linux-amd64-longtest](https://build.golang.org/log/0a9da5b62833bc0669b9145396289b55daf63f7f)
[2022-05-20T23:05:38-c3470ca/windows-amd64-longtest](https://build.golang.org/log/26e9a939f3d45f3023bc61c7abfd14f4b04e8193)
[2022-05-20T22:30:37-2b0e457/linux-amd64-longtest](https://build.golang.org/log/d339f424a273a3fa195237b2bc8050c93b7a7b66)
[2022-05-20T22:30:37-2b0e457/windows-amd64-longtest](https://build.golang.org/log/eae899f65c6331c5009f9f86a1c674fc3e432d94)
[2022-05-20T21:54:20-b580670/linux-amd64-longtest](https://build.golang.org/log/a0486e55a5a4f24d45c023465810aa8b81f1b848)
[2022-05-20T21:54:20-b580670/windows-amd64-longtest](https://build.golang.org/log/f6d48bf261c7003942c961c019fb674e600f77bc)
[2022-05-20T21:32:57-7ec6ef4/linux-amd64-longtest](https://build.golang.org/log/a3d9957bf3967e9017e4fc86cd495a26147c3499)
[2022-05-20T21:32:57-7ec6ef4/windows-amd64-longtest](https://build.golang.org/log/b9edaffffe9265e7aa6e1ace6ce0764431db1322)
[2022-02-03T22:55:24-248ad85/netbsd-amd64-9_0](https://build.golang.org/log/a9a3e653d41819d4031a8b64d90bc03d1770a522)

**Comment by bcmills** [comment]
This is some script test hanging while waiting on the subprocess's output pipes to close, with the output pipes stuck open.

This is quite possibly a bug in the `go` command, but it will be much more straightforward to diagnose once #50436 is implemented. (A workaround in the interim is possible but more complex, because we would either have to run our own pipe-copying goroutines or direct all of the `stderr` and `stdout` output pipes for script tests to files.)

I plan to implement #50436 for Go 1.20, so milestoning to Go 1.20.

**Comment by bcmills** [comment]
Duplicate of #50187.
